Pocket/pocket-ios
# Universal links - Add links to Widget Rows

## Summary
* This PR adds logic to send item links to the app in order to open them in the dedicated view

## References 
* Related widget Jira tickets

## Implementation Details
* Add `Link` modifier to `ItemWidgetRow`
* Add `Router` type to handle links

> [!IMPORTANT]
> the current implementation does not actually show the linked articles in the app yet. This will need to be added in a separate PR
> Unit tests will also need to be added here.

## Test Steps
* Put a breakpoint inside `Router` after the `guard` statement succeeds
* Install one instance of Saved Item or Recommendation widget, if you don't already have one
* tap on an article, and make sure the breakpoint gets hit

## PR Checklist:

## Screenshots
